app/src/main/java/ch/epfl/sdp/blindly/match/UserListFilter.kt

```diff
@@ -57,8 +57,8 @@ class UserListFilter {
         if (user.location == null || otherUser.location == null) {
             throw IllegalArgumentException("Cannot perform distance between null locations")
         }
-        val location = createLocation(user.location)
-        val otherLocation = createLocation(otherUser.location)
+        val location = createLocation(user.location!!)
+        val otherLocation = createLocation(otherUser.location!!)
```

Comment:
Are the `!!` really needed there ? There is an exception throw to get rid of the null values already.